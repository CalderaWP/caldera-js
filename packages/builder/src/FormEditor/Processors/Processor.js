import React, { Fragment, Component } from 'react';
import { TabPanel } from '@wordpress/components';
import {
	Row,
	Column,
	InputField,
	FORM_FIELDS_AUTO_COMPLETE_FIELD_TYPE_IDENTIFIER,
} from '@calderajs/components';
import { HorizontalForm } from '@calderajs/forms';
import PropTypes from 'prop-types';
import { AUTO_COMPLETE_FIELD_TYPE_IDENTIFIER } from '@calderajs/components';

const ProcessorLabel = ({ label }) => <div>{label}</div>;

const ProcessorLabelEdit = ({ label, onChange, id }) => (
	<InputField
		label={'Processor Label'}
		value={label}
		onChange={onChange}
		html5type={'text'}
		fieldId={`processor-label-${id}`}
	/>
);

const SettingsPanel = ({
	label,
	id,
	type,
	formProps,
	onEditLabel,
}) => {
	return (
		<div>
			<ProcessorLabel label={label ? label : type} />
			<ProcessorLabelEdit label={label} onChange={onEditLabel} id={id} />
			<HorizontalForm
				{...formProps}
				className={'caldera-processor-settings'}
			/>
		</div>
	);
};

export class Processor extends Component {
	state = {
		activeTab: 'settings',
	};

	onSetTab = activeTab => {
		this.setState({ activeTab });
	};

	tabs = [
		{
			name: 'settings',
			title: 'Settings',
			className: 'caldera-processor-settings-tab-btn',
		},
		{
			name: 'conditionals',
			title: 'Conditionals',
			className: 'caldera-processor-conditionals-tab-btn',
		},
	];

	/**
	 * Prepare props for the processor editing form
	 *
	 * @return {Processor.props}
	 */
	formProps = () => {
		const { form } = this.props;
		let props = { ...this.props };
		props.fields.map(field => {
			if (
				FORM_FIELDS_AUTO_COMPLETE_FIELD_TYPE_IDENTIFIER ===
				field.fieldType
			) {
				field.form = form;
			}
			return field;
		});
		return props;
	};

	onEditLabel = label => {
		this.props.onChange({ label });
	};



	render() {
		console.log(this.props.conditonalPanel);
		const {
			onClose,
			onRemove,
			conditionalPanel,
		} = this.props;
		return (
			<Fragment>
				<Row>
					<TabPanel
						className="caldera-processor"
						activeClass="active-tab"
						onSelect={this.onSetTab}
						initialTabName={'settings'}
						tabs={
							this.props.conditonalPanel
								? this.tabs
								: [this.tabs[0]]
						}
					>
						{() => {
							const { activeTab } = this.state;
							const tab = this.tabs.find(
								t => t.name === activeTab
							);
							const { name } = tab;
							if ('settings' === name) {
								return (
									<SettingsPanel
										{...this.props}
										formProps={this.formProps()}
										onEditLabel={this.onEditLabel}
									/>
								);
							}
							return (
								<div
									className={'caldera-processor-conditionals'}
								>
									{conditionalPanel}
								</div>
							);
						}}
					</TabPanel>
				</Row>
				<Row>
					<Column width={'1/2'}>
						<button
							className={'caldera-processor-close'}
							onClick={onClose}
						>
							Close
						</button>
					</Column>
					<Column width={'1/2'}>
						<button
							className={'caldera-processor-remove'}
							onClick={onRemove}
						>
							Remove
						</button>
					</Column>
				</Row>
			</Fragment>
		);
	}
}

Processor.propTypes = {
	...HorizontalForm.propTypes,
	label: PropTypes.string,
	initialActiveTab: PropTypes.string,
	onRemove: PropTypes.func,
	type: PropTypes.string,
	id: PropTypes.string,
	form: PropTypes.shape({
		id: PropTypes.string.isRequired,
		fields: PropTypes.array.isRequired,
	}).isRequired,
};

Processor.defaultProps = {
	...HorizontalForm.defaultProps,
	initialActiveTab: 'settings',
	conditionalPanel: null,
};
