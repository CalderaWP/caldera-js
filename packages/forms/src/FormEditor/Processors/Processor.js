import React, { Fragment, Component } from 'react';
import { HorizontalForm } from '../../HorizontalForm/HorizontalForm';
import { TabPanel } from '@wordpress/components';
import { Row, Column,FormFieldsAutoComplete } from"@calderawp/components";
import PropTypes from 'prop-types';

export class Processor extends Component {
	state = {
		activeTab: 'settings'
	};

	onSetTab = activeTab => {
		this.setState({ activeTab });
	};

	tabs = [
		{
			name: 'settings',
			title: 'Settings',
			className: 'caldera-processor-settings-tab-btn'
		},
		{
			name: 'conditionals',
			title: 'Conditionals',
			className:
				'caldera-processor-conditionals-tab-btn'
		}
	];

	/**
	 * Prepare props for the processor editing form
	 *
	 * @return {Processor.props}
	 */
	formProps = () => {
		let {props} = this;
		const {fields,form} = this.props;
		const formFields = form.fields;
		fields.forEach( (field,i) => {
			const {fieldType} = field;
			if( FormFieldsAutoComplete.IDENTIFIER === fieldType ){
				props.fields[i] = {
					...field,
					form
				}
			}
		});
		return props;
	};

	render() {
		const { onClose, onRemove } = this.props;
		return (
			<Fragment>
				<Row>
					<TabPanel
						className="caldera-processor"
						activeClass="active-tab"
						onSelect={this.onSetTab}
						initialTabName={'settings'}
						tabs={this.tabs}
					>
						{() => {
							const {activeTab} = this.state;
							const tab =  this.tabs.find( t => t.name === activeTab );
							const { name } = tab;
							if ('settings' === name) {
								return (
									<HorizontalForm
										{...this.formProps()}
										className={'caldera-processor-settings'}
									/>
								);
							}
							return (
								<div
									className={'caldera-processor-conditionals'}
								>
									Conditionals
								</div>
							);
						}}
					</TabPanel>
				</Row>
				<Row>
					<Column width={'1/2'}>
						<button
							className={'caldera-processor-close'}
							onClick={onClose}>
							Close
						</button>
					</Column>
					<Column width={'1/2'}>
						<button
							className={'caldera-processor-remove'}
							onClick={onRemove}>
							Remove
						</button>
					</Column>
				</Row>
			</Fragment>
		);
	}
}

//should have label!
Processor.propTypes = {
	...HorizontalForm.propTypes,
	initialActiveTab: PropTypes.string,
	onRemove: PropTypes.func,
	type: PropTypes.string,
	id: PropTypes.string,
};

Processor.defaultProps = {
	...HorizontalForm.defaultProps,
	initialActiveTab: 'settings'
};
