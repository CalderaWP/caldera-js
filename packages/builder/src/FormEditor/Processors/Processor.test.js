import * as React from "react";
import { mount } from "enzyme";
import { render } from "react-testing-library";
import { Processor } from "./Processor";

import {
	checkboxFieldset,
	selectField,
	checkboxField,
} from "@calderajs/components";

describe("Processor", () => {
	let onClose;
	let onChange;
	let onBlur;

	const form = {
		ID: "cf1",
		fields: [checkboxFieldset, selectField, checkboxField],
	};

	beforeEach(() => {
		onClose = jest.fn();
		onChange = jest.fn();
		onBlur = jest.fn();
	});

	it("Matches snapshot", () => {
		const component = render(
			<Processor
				fields={[checkboxFieldset, selectField, checkboxField]}
				initialValues={{}}
				onClose={onClose}
				onChange={onChange}
				onBlur={onBlur}
				instanceId={"test-1"}
				form={form}
			/>
		);
		expect(component).toMatchSnapshot();
	});

	it("Can switch to conditionals", () => {
		const component = mount(
			<Processor
				fields={[checkboxFieldset, selectField, checkboxField]}
				initialValues={{}}
				onClose={onClose}
				onChange={onChange}
				onBlur={onBlur}
				initialActiveTab={"conditionals"}
				instanceId={"test-1"}
				form={form}
			/>
		);
		component.instance().onSetTab("conditionals");
		expect(component.state("activeTab")).toBe("conditionals");
	});

	it("Can show conditionals", () => {
		const component = mount(
			<Processor
				fields={[checkboxFieldset, selectField, checkboxField]}
				initialValues={{}}
				onClose={onClose}
				onChange={onChange}
				onBlur={onBlur}
				initialActiveTab={"conditionals"}
				instanceId={"test-1"}
				form={form}
			/>
		);
		expect(component.find(".caldera-processor-conditionals").length).toBe(
			0
		);
		component.setState({ activeTab: "conditionals" });
		expect(component.state("activeTab")).toBe("conditionals");
	});
});
