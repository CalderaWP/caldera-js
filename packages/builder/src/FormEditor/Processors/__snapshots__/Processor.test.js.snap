// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Processor Matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-polite"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions text"
      class="a11y-speak-region"
      id="a11y-speak-assertive"
      style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
    />
    <div>
      <div
        class="sc-bdVaJa caldera-row sc-bwzfXH pjOvZ"
      >
        <div
          class="caldera-processor"
        >
          <div
            aria-orientation="horizontal"
            class="components-tab-panel__tabs"
            role="tablist"
          >
            <button
              aria-controls="0-settings-view"
              aria-selected="true"
              class="components-button caldera-processor-settings-tab-btn active-tab"
              id="0-settings"
              role="tab"
              type="button"
            >
              Settings
            </button>
            <button
              aria-controls="0-conditionals-view"
              aria-selected="false"
              class="components-button caldera-processor-conditionals-tab-btn "
              id="0-conditionals"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Conditionals
            </button>
          </div>
          <div
            aria-labelledby="0-settings"
            class="components-tab-panel__tab-content"
            id="0-settings-view"
            role="tabpanel"
            tabindex="0"
          >
            <div
              class="caldera"
            >
              <form
                class="horizontal-form-test-1 caldera-form"
              >
                <div
                  class="sc-bdVaJa caldera-form-row caldera-row sc-bwzfXH pjOvZ"
                >
                  <div
                    class="caldera-column width-1 sc-bdVaJa iHZvIS"
                    id="c-test-1-0"
                  >
                    <div
                      class="caldera-field-group caldera-field-area-checkboxes is-required form-group"
                    >
                      <label
                        class="form-label"
                        for="checkboxFieldSetLabel"
                      >
                        Checkbox Field Set Label
                         
                        <span>
                          *
                        </span>
                      </label>
                      <fieldset>
                        <legend>
                          Checkbox Field Set Label
                        </legend>
                        <div
                          class="caldera-field caldera-field-checkbox form-check"
                        >
                          <input
                            class="form-check-input"
                            id="opt-1"
                            type="checkbox"
                          />
                          <label
                            class="form-check-label"
                            for="opt-1"
                            title=""
                            type="checkbox"
                          >
                            One
                          </label>
                        </div>
                        <div
                          class="caldera-field caldera-field-checkbox form-check"
                        >
                          <input
                            class="form-check-input"
                            id="opt-2"
                            type="checkbox"
                          />
                          <label
                            class="form-check-label"
                            for="opt-2"
                            title=""
                            type="checkbox"
                          >
                            Two
                          </label>
                        </div>
                        <div
                          class="caldera-field caldera-field-checkbox form-check"
                        >
                          <input
                            class="form-check-input"
                            id="opt-3"
                            type="checkbox"
                          />
                          <label
                            class="form-check-label"
                            for="opt-3"
                            title=""
                            type="checkbox"
                          >
                            Three
                          </label>
                        </div>
                      </fieldset>
                      <small
                        class="text-muted form-text"
                      >
                        Checkbox field set description
                      </small>
                    </div>
                  </div>
                </div>
                <div
                  class="sc-bdVaJa caldera-form-row caldera-row sc-bwzfXH pjOvZ"
                >
                  <div
                    class="caldera-column width-1 sc-bdVaJa iHZvIS"
                    id="c-test-1-1"
                  >
                    <div
                      class="caldera-field-group caldera-field-area-select is-required form-group"
                    >
                      <label
                        class="form-label"
                        for="selectFieldId"
                      >
                        Select Field Label
                         
                        <span>
                          *
                        </span>
                      </label>
                      <select
                        class="caldera-field caldera-field-select form-control"
                        id="selectFieldId"
                        required=""
                      >
                        <option>
                          --
                        </option>
                        <option
                          value="0"
                        >
                          Zero
                        </option>
                        <option
                          value="1"
                        >
                          One
                        </option>
                        <option
                          value="2"
                        >
                          Two
                        </option>
                      </select>
                      <small
                        class="text-muted form-text"
                      >
                        Select field description
                      </small>
                    </div>
                  </div>
                </div>
                <div
                  class="sc-bdVaJa caldera-form-row caldera-row sc-bwzfXH pjOvZ"
                >
                  <div
                    class="caldera-column width-1 sc-bdVaJa iHZvIS"
                    id="c-test-1-2"
                  >
                    <div
                      class="caldera-field-group caldera-field-area-checkbox is-required form-group"
                    >
                      <label
                        class="form-label"
                        for="checkboxFieldLabel"
                      >
                        Checkbox Labe
                         
                        <span>
                          *
                        </span>
                      </label>
                      <div
                        class="caldera-field caldera-field-checkbox form-check"
                      >
                        <input
                          attributes="[object Object]"
                          class="form-check-input"
                          id="checkboxFieldLabel"
                          required=""
                          type="checkbox"
                        />
                        <label
                          class="form-check-label"
                          for="checkboxFieldLabel"
                          title=""
                          type="checkbox"
                        >
                          Checkbox Labe
                        </label>
                      </div>
                      <small
                        class="text-muted form-text"
                      >
                        Checkbox field description
                      </small>
                    </div>
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-bdVaJa caldera-row sc-bwzfXH pjOvZ"
      >
        <div
          class="caldera-column width-6 sc-bdVaJa gfmZBt"
          width="1/2"
        >
          <button
            class="caldera-processor-close"
          >
            Close
          </button>
        </div>
        <div
          class="caldera-column width-6 sc-bdVaJa gfmZBt"
          width="1/2"
        >
          <button
            class="caldera-processor-remove"
          >
            Remove
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdVaJa caldera-row sc-bwzfXH pjOvZ"
    >
      <div
        class="caldera-processor"
      >
        <div
          aria-orientation="horizontal"
          class="components-tab-panel__tabs"
          role="tablist"
        >
          <button
            aria-controls="0-settings-view"
            aria-selected="true"
            class="components-button caldera-processor-settings-tab-btn active-tab"
            id="0-settings"
            role="tab"
            type="button"
          >
            Settings
          </button>
          <button
            aria-controls="0-conditionals-view"
            aria-selected="false"
            class="components-button caldera-processor-conditionals-tab-btn "
            id="0-conditionals"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Conditionals
          </button>
        </div>
        <div
          aria-labelledby="0-settings"
          class="components-tab-panel__tab-content"
          id="0-settings-view"
          role="tabpanel"
          tabindex="0"
        >
          <div
            class="caldera"
          >
            <form
              class="horizontal-form-test-1 caldera-form"
            >
              <div
                class="sc-bdVaJa caldera-form-row caldera-row sc-bwzfXH pjOvZ"
              >
                <div
                  class="caldera-column width-1 sc-bdVaJa iHZvIS"
                  id="c-test-1-0"
                >
                  <div
                    class="caldera-field-group caldera-field-area-checkboxes is-required form-group"
                  >
                    <label
                      class="form-label"
                      for="checkboxFieldSetLabel"
                    >
                      Checkbox Field Set Label
                       
                      <span>
                        *
                      </span>
                    </label>
                    <fieldset>
                      <legend>
                        Checkbox Field Set Label
                      </legend>
                      <div
                        class="caldera-field caldera-field-checkbox form-check"
                      >
                        <input
                          class="form-check-input"
                          id="opt-1"
                          type="checkbox"
                        />
                        <label
                          class="form-check-label"
                          for="opt-1"
                          title=""
                          type="checkbox"
                        >
                          One
                        </label>
                      </div>
                      <div
                        class="caldera-field caldera-field-checkbox form-check"
                      >
                        <input
                          class="form-check-input"
                          id="opt-2"
                          type="checkbox"
                        />
                        <label
                          class="form-check-label"
                          for="opt-2"
                          title=""
                          type="checkbox"
                        >
                          Two
                        </label>
                      </div>
                      <div
                        class="caldera-field caldera-field-checkbox form-check"
                      >
                        <input
                          class="form-check-input"
                          id="opt-3"
                          type="checkbox"
                        />
                        <label
                          class="form-check-label"
                          for="opt-3"
                          title=""
                          type="checkbox"
                        >
                          Three
                        </label>
                      </div>
                    </fieldset>
                    <small
                      class="text-muted form-text"
                    >
                      Checkbox field set description
                    </small>
                  </div>
                </div>
              </div>
              <div
                class="sc-bdVaJa caldera-form-row caldera-row sc-bwzfXH pjOvZ"
              >
                <div
                  class="caldera-column width-1 sc-bdVaJa iHZvIS"
                  id="c-test-1-1"
                >
                  <div
                    class="caldera-field-group caldera-field-area-select is-required form-group"
                  >
                    <label
                      class="form-label"
                      for="selectFieldId"
                    >
                      Select Field Label
                       
                      <span>
                        *
                      </span>
                    </label>
                    <select
                      class="caldera-field caldera-field-select form-control"
                      id="selectFieldId"
                      required=""
                    >
                      <option>
                        --
                      </option>
                      <option
                        value="0"
                      >
                        Zero
                      </option>
                      <option
                        value="1"
                      >
                        One
                      </option>
                      <option
                        value="2"
                      >
                        Two
                      </option>
                    </select>
                    <small
                      class="text-muted form-text"
                    >
                      Select field description
                    </small>
                  </div>
                </div>
              </div>
              <div
                class="sc-bdVaJa caldera-form-row caldera-row sc-bwzfXH pjOvZ"
              >
                <div
                  class="caldera-column width-1 sc-bdVaJa iHZvIS"
                  id="c-test-1-2"
                >
                  <div
                    class="caldera-field-group caldera-field-area-checkbox is-required form-group"
                  >
                    <label
                      class="form-label"
                      for="checkboxFieldLabel"
                    >
                      Checkbox Labe
                       
                      <span>
                        *
                      </span>
                    </label>
                    <div
                      class="caldera-field caldera-field-checkbox form-check"
                    >
                      <input
                        attributes="[object Object]"
                        class="form-check-input"
                        id="checkboxFieldLabel"
                        required=""
                        type="checkbox"
                      />
                      <label
                        class="form-check-label"
                        for="checkboxFieldLabel"
                        title=""
                        type="checkbox"
                      >
                        Checkbox Labe
                      </label>
                    </div>
                    <small
                      class="text-muted form-text"
                    >
                      Checkbox field description
                    </small>
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-bdVaJa caldera-row sc-bwzfXH pjOvZ"
    >
      <div
        class="caldera-column width-6 sc-bdVaJa gfmZBt"
        width="1/2"
      >
        <button
          class="caldera-processor-close"
        >
          Close
        </button>
      </div>
      <div
        class="caldera-column width-6 sc-bdVaJa gfmZBt"
        width="1/2"
      >
        <button
          class="caldera-processor-remove"
        >
          Remove
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
